core/predicate_check.go

```diff
@@ -24,46 +24,15 @@ func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.Prop
 	if tx.Gas() < intrinsicGas {
 		return fmt.Errorf("insufficient gas for predicate verification (%d) < intrinsic gas (%d)", tx.Gas(), intrinsicGas)
 	}
-	if err := checkPrecompilePredicates(rules, &predicateContext.PrecompilePredicateContext, tx); err != nil {
-		return err
-	}
-	return checkProposerPrecompilePredicates(rules, predicateContext, tx)
-}
-
-func checkPrecompilePredicates(rules params.Rules, predicateContext *precompileconfig.PrecompilePredicateContext, tx *types.Transaction) error {
-	// Short circuit early if there are no precompile predicates to verify
-	if len(rules.PredicatePrecompiles) == 0 {
-		return nil
-	}
-	precompilePredicates := rules.PredicatePrecompiles
-	// Track addresses that we've performed a predicate check for
-	precompileAddressChecks := make(map[common.Address]struct{})
-	for _, accessTuple := range tx.AccessList() {
-		address := accessTuple.Address
-		predicater, ok := precompilePredicates[address]
-		if !ok {
-			continue
-		}
-		// Return an error if we've already checked a predicate for this address
-		if _, ok := precompileAddressChecks[address]; ok {
-			return fmt.Errorf("predicate %s failed verification for tx %s: specified %s in access list multiple times", address, tx.Hash(), address)
-		}
-		precompileAddressChecks[address] = struct{}{}
-		predicateBytes := predicateutils.HashSliceToBytes(accessTuple.StorageKeys)
-		if err := predicater.VerifyPredicate(predicateContext, predicateBytes); err != nil {
-			return fmt.Errorf("predicate %s failed verification for tx %s: %w", address, tx.Hash(), err)
-		}
-	}
-
-	return nil
+	return checkPrecompilePredicates(rules, predicateContext, tx)
 }
 
-func checkProposerPrecompilePredicates(rules params.Rules, predicateContext *precompileconfig.ProposerPredicateContext, tx *types.Transaction) error {
+func checkPrecompilePredicates(rules params.Rules, predicateContext *precompileconfig.PredicateContext, tx *types.Transaction) error {
 	// Short circuit early if there are no precompile predicates to verify
-	if len(rules.ProposerPredicates) == 0 {
+	if len(rules.Predicates) == 0 {
 		return nil
 	}
-	precompilePredicates := rules.ProposerPredicates
+	precompilePredicates := rules.Predicates
```

Comment:
nit: could we get rid of this variable (or rename to predicates, but doesn't seem very helpful)?